packages/back-end/src/api/dimensions/postDimension.ts

```diff
@@ -0,0 +1,41 @@
+import uniqid from "uniqid";
+import { PostDimensionResponse } from "back-end/types/openapi";
+import { createApiRequestHandler } from "back-end/src/util/handler";
+import {
+  createDimension,
+  toDimensionApiInterface,
+} from "back-end/src/models/DimensionModel";
+import { getDataSourceById } from "back-end/src/models/DataSourceModel";
+import { postDimensionValidator } from "back-end/src/validators/openapi";
+
+export const postDimension = createApiRequestHandler(postDimensionValidator)(
+  async (req): Promise<PostDimensionResponse> => {
+    const organization = req.organization.id;
+
+    const datasourceDoc = await getDataSourceById(
+      req.context,
+      req.body.datasourceId
+    );
+    if (!datasourceDoc) {
+      throw new Error("Invalid data source");
+    }
+
```

Comment:
Updating the enum options might fix this itself, but I wonder if we want to add a safe guard to this endpoint (and the update endpoint) to return an error if the payload's `managedBy` property is set to `"config"`, given how the user would then be unable to edit/delete the Dimension.

It may be that we get this for free by updating the yaml, but not sure without testing.

Reply:
The zod validator should give us this for free (given the yaml fix you suggested)